src/configobj/__init__.py

```diff
@@ -1198,8 +1198,15 @@ def __init__(self, infile=None, options=None, configspec=None, encoding=None,
         self._original_configspec = configspec
         self._load(infile, configspec)
 
-
     def _load(self, infile, configspec):
+        if six.PY3:
+            import pathlib
+            if isinstance(infile, pathlib.Path):
+                infile = str(infile)
+            # This is basically os.fspath, but backwards-compatible
+            if hasattr(infile, '__fspath__'):
+                infile = infile.__fspath__()
+
```

Comment:
These are all rather specific tests (including the existing ones). It'd probably be better to use good old duck-typing, i.e. a nested ``try: infile + '' … try: infile / '' …``.